### DISCUSSION TITLE
How do we do.... anything other than extremely basic stuff?

### DISCUSSION THREAD

**Discussion by amuhareb** [langflow-ai/langflow#170] [Q&A]
hi! As a noobie to AI and especially tools like langchain, this appeals to me deeply. However, there is a severe lack of clarity in that there's no explanation about what's expected of the user nor is there any documentation to speak of. In my attempts(in vain) to integrate tools with the LLM today, I could not figure out at all what the proper method for getting agents to actually deploy tools is. I was fully under the assumption that a python script block would simply run, print the results to the output and passed to the agent along with the prompt. But, beyond being completely in the dark about how to actually set up such a flow(why would a python tool need to connect to a tool? is this not severely redundant? why is there no input for the python object?) I could not for the life of me parse the Langchain documentation on prompts in a way that had meaningful implications for your GUI - do i need to call imports in the GUI? Is the format actually python, or something else? Why would the prompt, which is concurrent separate to the 'tool' object, rely on the user to include an outside source that is concurrent? Would it not make more sense for there to be some sort of 'combiner' object that links those two flows up? 

I know you said you're focusing on getting as many modules in as you can, but for the sake of users, I hope you also spend some time on example flows and documentation.

I know this is in it's infancy, and i'm excited to see where it goes, and I hope to contribute in any way I can. I just spent all day playing around with this and I left feeling vastly more confused than when i went in.

Cheers.

**Comment by clstaudt** [comment]
There are now a few examples that can be imported.

![Screenshot 2023-05-03 at 22 26 08](https://user-images.githubusercontent.com/875194/236041751-efcc3226-7098-43c7-9896-0dc3bab229cc.png)

I suppose they serve as starting points for more than extremely basic stuff. The problem here is still the lack of documentation. The examples are not self-explanatory and are missing a text description. Tutorial-style examples would be great.

**Reply by Lutios** [reply]
Thanks for the tip, but how do you bring up this "Select an Example" menu? I've been staring at the UI and clicking on things and can't see anything that would bring up such a menu.

**Comment by mike12432** [comment]
top right under import

**Comment by rodrigosnader** [comment]
Hi @amuhareb, I hope this is getting better with the new version. Don't hesitate to reach out on our [discord](http://discord.gg/EqksyE2EX9) channel. We'll do our best to help :)

**Comment by carlosrcoelho** [comment]
Hey!

Thanks for sharing your question here. I'm sorry for not getting back to you sooner.
Langflow has gone through some major changes since then. I recommend trying out the new version for the best experience. If you have any issues, please open a new one; we'll be here to help.
